spaceship/lib/spaceship/portal/app.rb

```diff
@@ -74,12 +133,21 @@ def factory(attrs)
           self.new(attrs)
         end
 
+        # <b>DEPRECATED:</b> Use <tt>all_by_platform</tt> instead.
         # @param mac [Bool] Fetches Mac apps if true
         # @return (Array) Returns all apps available for this account
         def all(mac: false)
-          client.apps(mac: mac).map { |app| self.factory(app) }
+          puts '`all` is deprecated. Please use `all_by_platform` instead.'.red
+          all_by_platform(platform: mac ? Spaceship::Portal::App::MAC : Spaceship::Portal::App::IOS)
+        end
+
+        # @param platform (String) The platform to search for
+        # @return (Array) Returns all apps available for this account
+        def all_by_platform(platform: nil)
```

Comment:
Since this results in much more code for the user
```ruby
Spaceship::Application.all
# VS
Spaceship::Application.all_by_platform(platform: Spaceship::Portal::App::IOS)
```

how about we add convenience methods like

```ruby
Spaceship::Application.all_ios
Spaceship::Application.all_mac
```
which will use `all_by_platform` under the hood